File: src/test/java/lsd/format/xml/XmlPrettyPrinterShould.java
package lsd.format.xml;

import org.junit.jupiter.api.Test;

import java.util.Optional;

import static java.util.Optional.empty;
import static org.hamcrest.MatcherAssert.assertThat;
import static org.hamcrest.Matchers.is;
import static org.junit.jupiter.api.Assertions.assertTrue;

public class XmlPrettyPrinterShould {

    @Test
    public void indentXml() {
        final Optional<String> result = XmlPrettyPrinter.indentXml("<?xml version=\"1.0\" encoding=\"UTF-8\"?><parent attribute=\"abc\"><child>value</child></parent>");
        assertTrue(result.isPresent());
        assertThat(result.get(), is("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n\n<parent attribute=\"abc\">\n  <child>value</child>\n</parent>\n"));
    }

    @Test
    public void handleEmptyString() {
        final Optional<String> result = XmlPrettyPrinter.indentXml(null);
        assertThat(result, is(empty()));
    }

    @Test
    public void handleNonXml() {
        final Optional<String> result = XmlPrettyPrinter.indentXml("blah");
        assertThat(result, is(empty()));
    }
}

File: src/test/java/lsd/format/xml/XmlPrettyPrinterShould.kt
package lsd.format.xml

import org.hamcrest.MatcherAssert.assertThat
import org.hamcrest.Matchers.*
import org.junit.jupiter.api.Test

class XmlPrettyPrinterShould {
    @Test
    fun `indent xml`() {
        val result =
            indentXml("<?xml version=\"1.0\" encoding=\"UTF-8\"?><parent attribute=\"abc\"><child>value</child></parent>")
        assertThat(result, notNullValue())
        assertThat(
            result,
            `is`("<?xml version=\"1.0\" encoding=\"UTF-8\"?>\n\n<parent attribute=\"abc\">\n  <child>value</child>\n</parent>\n")
        )
    }

    @Test
    fun `handle empty string`() {
        val result = indentXml(null)
        assertThat(result, `is`(nullValue()))
    }

    @Test
    fun `handle non xml`() {
        val result = indentXml("blah")
        assertThat(result, `is`(nullValue()))
    }
}